simulations/vip-converter/vip-converter-testnet/simulations.ts

```diff
@@ -0,0 +1,345 @@
+    psrMaxLoopsLimit = await ProtocolShareReserve.maxLoopsLimit();
+    psrTotalAssetReserve = await ProtocolShareReserve.totalAssetReserve(XVS);
+  });
+
+  describe("Pre-VIP behavior Check for RiskFund and PSR", () => {
+    it("RiskFund Proxy should have old implementation", async () => {
+      expect(await ProxyAdmin.getProxyImplementation(RISK_FUND_PROXY)).to.equal(RISK_FUND_OLD_IMPLEMENTATION);
+    });
+
+    it("ProtocolShareReserve Proxy should have old implementation", async () => {
+      expect(await ProxyAdmin.getProxyImplementation(PROTOCOL_SHARE_RESERVE_PROXY)).to.equal(
+        PROTOCOL_SHARE_RESERVE_OLD_IMPLEMENTATION,
+      );
+    });
+  });
+
+  testVip("VIP-converter", vipConverter(), {
+    callbackAfterExecution: async (txResponse: any) => {
+      await expectEvents(
+        txResponse,
+        [SINGLE_TOKEN_CONVERTER_ABI],
+        ["OwnershipTransferred", "ConverterNetworkAddressUpdated"],
+        [8, 6],
+      );
+      await expectEvents(txResponse, [CONVERTER_NETWORK_ABI], ["ConverterAdded"], [6]);
+      await expectEvents(
+        txResponse,
+        [PROTOCOL_SHARE_RESERVE_OLD_IMPLEMENTATION_ABI],
+        ["DistributionConfigUpdated", "DistributionConfigRemoved"],
+        [4, 4],
+      );
+      await expectEvents(
+        txResponse,
+        [PROTOCOL_SHARE_RESERVE_NEW_IMPLEMENTATION_ABI],
+        ["DistributionConfigAdded"],
+        [10],
+      );
+      await expectEvents(txResponse, [ACCESS_CONTROL_MANAGER_ABI], ["PermissionGranted"], [28]);
+      await expectEvents(txResponse, [RISK_FUND_V2_ABI], ["RiskFundConverterUpdated"], [1]);
+      await expectEvents(txResponse, [TRANSPARENT_PROXY_ABI], ["Upgraded"], [2]);
+      await expectEvents(txResponse, [SINGLE_TOKEN_CONVERTER_ABI], ["ConversionConfigUpdated"], [210]);
+    },
+  });
+
+  describe("Post-VIP behavior", () => {
+    before(() => {
+      RiskFund = new ethers.Contract(RISK_FUND_PROXY, RISK_FUND_V2_ABI, provider);
+      ProtocolShareReserve = new ethers.Contract(
+        PROTOCOL_SHARE_RESERVE_PROXY,
+        PROTOCOL_SHARE_RESERVE_NEW_IMPLEMENTATION_ABI,
+        provider,
+      );
+    });
+
+    it("RiskFund Proxy should have new implementation", async () => {
+      expect(await ProxyAdmin.getProxyImplementation(RISK_FUND_PROXY)).to.equal(RISK_FUND_V2_IMPLEMENTATION);
+    });
+
+    it("RiskFund should have correct storage", async () => {
+      expect(await RiskFund.accessControlManager()).to.equal(ACM);
+      expect(await RiskFund.convertibleBaseAsset()).to.equal(riskFundConvertibleBaseAsset);
+      expect(await RiskFund.maxLoopsLimit()).to.equal(riskFundMaxLoopsLimit);
+      expect(await RiskFund.owner()).to.equal(NORMAL_TIMELOCK);
+      expect(await RiskFund.shortfall()).to.equal(riskFundShortfall);
+    });
+
+    it("RiskFund should have correct address of RiskFundConverter", async () => {
+      expect(await RiskFund.riskFundConverter()).to.equal(RISK_FUND_CONVERTER);
+    });
+
```

Comment:
Regarding the RiskFund, I would also check the funds held by the contracts (the balances) allocated to the different comptroller-asset

Reply:
Done https://github.com/VenusProtocol/vips/pull/49/commits/0473a2acadf59f75370bf39aa959d6cd9a55476e

Reply:
Done https://github.com/VenusProtocol/vips/pull/49/commits/9eb73c07ce3c22d0e06dac9f5ba757469d673f1c